### PR TITLE
Fix bug crash relaunch on Android

### PR DESCRIPTION
**Pull request by Levis-ScopicSoftware**
Remove load lazy activity and local broadcast that is cause retain instance of activity when press back and relaunch app.

### CODE REVIEW

File: react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt [hunk 1]
```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
...
-          getCurrentLocation()
-        }
-        waitingForCurrentLocationPermission = false
-      },
-      exceptionCallback = {
-        waitingForLocationPermission = false
-      })
-
-    return true
+    if (currentActivity != null) {
```
Comment (RSMNYS) [inline react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt]: Please don't do many levels of if. It's hard to understand. Better to have the structure more flat. If (activity === null) {return false}. And so on

File: react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt [hunk 2]
```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
...
+        builder.updateInterval(it)
       }
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_OK_BTN)?.let {
-        permissionConfigBuilder.okButton(it)
+      getIntOrNull(params, KEY_MAX_LOCATION_ACCURACY)?.let {
+        builder.maximumLocationAccuracy(it)
       }
-      getStringOrNull(permissionRequestParams, KEY_PERMISSION_REQUEST_CANCEL_BTN)?.let {
-        permissionConfigBuilder.cancelButton(it)
+      getMapOrNull(params, KEY_PERMISSION_REQUEST_CONFIG)?.let { permissionRequestParams ->
```
Comment (RSMNYS) [inline react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt]: here we can extract permissionConfigBuilder into a separate function with all configs.

File: react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt [hunk 3]
```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
...
   }
 
   @ReactMethod
   fun startLocationUpdates() {
-    SYRFLocation.subscribeToLocationUpdates(activity) { _, error ->
-      if (error != null) {
-        if (error is MissingLocationException) {
-          waitingForLocationPermission = true
-          requestLocationPermission()
+    if (currentActivity != null) {
```
Comment (RSMNYS) [inline react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt]: Same here with the flat structure.

File: react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt [hunk 4]
```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
...
   }
 
   @ReactMethod
   fun getCurrentLocation() {
-    SYRFLocation.getCurrentPosition(activity) { location, error ->
-      if (error != null) {
-        if (error is MissingLocationException) {
-          waitingForCurrentLocationPermission = true
-          requestLocationPermission()
+    if (currentActivity != null) {
```
Comment (RSMNYS) [inline react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt]: and here

**Review by RSMNYS — CHANGES_REQUESTED** [review]

**Comment by Levis-ScopicSoftware** [comment]
@RSMNYS  I have fixed comments, please help me check again. Thanks

**Comment by RSMNYS** [comment]
Just a question, why do you think it's crashing. Also have you tried your fix and make sure it behaves differently?

**Pull request merged**